src/site/content/en/learn/css/media-queries/index.md

```diff
@@ -0,0 +1,786 @@
+
+
+With CSS Media Queries you can solve this problem:
+
+{% Codepen {
+  user: 'web-dot-dev',
+  id: 'YzryWpw',
+  theme: 'dark',
+  height: 700,
+  tab: 'css, result'
+} %}
+
+{% YouTube "JLRFF56_S3U" %}
+
+Because the Web is agnostic, there became a need across standardized Web languages for Web developers to be able inquire whether or not a set of criteria applies to a medium. The result of that was the [Media Query]  standard, a normalized means to write a query with a `true` or `false` result whether certain media conditions hold true for a particular device the Web is being consumed on.
+
+CSS styles are applied to documents that can be consumed in a medium other than browsers; accordingly media queries are a handy way for Web creatives to conditionally apply styles to fix, enhance, or maximize a document's usefulness accounting for particular media conditions applicable
+
+In other words, keeping in mind `media` is plural for medium, CSS media queries allow you to apply styles originating from blocks of CSS or a stylesheet depending on medium-related conditions applicable to the device a user is viewing a particular document with. These media-related conditions include the capabilities you require a medium to have & even the users preferences being actively accommodated by the medium.
+
+In CSS you apply a media query using either the  `@media`  or `@import`  [at rules](https://thecsspodcast.libsyn.com/031-rules)  followed by the [**media query**](l).
+
+`@media` is used to conditionally apply styles based on a media query while `@link` conditionally loads a stylesheet based on a media query.
+
+Media queries immediately follow the `@media`   at-rule, while media queries follows the URL to a stylesheet & optionally feature queries when used with the `@import` at-rule.
+
+To begin the module we conditionally placed cards on a grid with the following code:
+
+```css
+#cards {
+  display: grid;
+  grid-template: 1fr;
+  grid-gap: 2rem;
+  width: 100%;
+  padding: 10%;
+  margin: 0;
+  list-style: none;
+  border: 2px solid magenta;
+}
+
+@media (min-width: 31.25em /* 500px */)  {
+  #cards {
+    grid-template-columns: 1fr 1fr;
+  }
+}
+
+@media (min-width: 65.8125em /* 1053px */)  {
+  #cards {
+    grid-template-columns: 1fr 1fr 1fr;
+  }
+}
+
+@media(min-width: 80em /* 1280px */) {
+  #cards {
+    grid-template-columns: repeat(auto-fill, minmax(20em, 1fr))
+  }
+}
+```
+
+The equivalent if the `@media` at-rule code blocks were co-located in a separate stylesheet  would be the following
+
+```css
+@import "meaningful-layout-tweaks-to-cards-on-larger-viewports.css" (min-width: 31.25em);
+```
+
+{% Aside 'warning' %}
+Because of performance reasons, `@import` should be utilized with caution. It often is a better choice to use `link` or concatenating the styles you considered importing instead.
```

Comment:
```suggestion
Because of performance reasons, `@import` should be utilized with caution. It often is a better choice to use `<link>` or concatenating the styles you considered importing instead.
```

Reply:
I'd briefly explain the performance reasons against `@import`. Also, this may give users of preprocessors the opinion that `@import` in those tools is similarly bad, but they're actually fine since they end up concatenating styles loaded from `@import` rules. It's a bit of a finicky problem space 😅